docs/reference/ssl/kubernetes-with-cert-manager.md

```diff
@@ -0,0 +1,203 @@
+
+```yaml
+# letsencrypt-issuer.yaml
+apiVersion: certmanager.k8s.io/v1alpha1
+kind: Issuer
+metadata:
+  name: letsencrypt-prod
+  namespace: openfaas
+spec:
+  acme:
+    # Email address used for ACME registration
+    email: <your-email-here>
+    http01: {}
+    # Name of a secret used to store the ACME account private key
+    privateKeySecretRef:
+      key: ""
+      name: letsencrypt-prod
+    server: https://acme-v01.api.letsencrypt.org/directory
+---
+apiVersion: certmanager.k8s.io/v1alpha1
+kind: Issuer
+metadata:
+  name: letsencrypt-staging
+  namespace: openfaas
+spec:
+  acme:
+    server: https://acme-staging.api.letsencrypt.org/directory
+    # Email address used for ACME registration
+    email: <your-email-here>
+    # Name of a secret used to store the ACME account private key
+    privateKeySecretRef:
+      name: letsencrypt-staging
+    http01: {}
+```
+
+```sh
+$ kubectl apply -f letsencrypt-issuer.yaml
+```
+
+This will allow `cert-manager` to automatically provision Certificates just in the `openfaas` namespace.
+
+## Add TLS to openfaas
+
+The OpenFaaS Helm Chart already supports the nginx-ingress, but we want to customize it further. This is easiest with a custom values file. Below, we enable and configure the ingress object to use our certificate and expose just the gateway
+
+```yaml
+# tls.yml
+ingress:
+    enabled: true
+    annotations:
+        kubernetes.io/ingress.class: nginx
+        certmanager.k8s.io/cluster-issuer: letsencrypt-staging
+    ingress:
+        tls:
+        - hosts:
+          - openfaas.mydomain.com
+          secretName: openfaas-crt
+        hosts:
+        - host: openfaas.mydomain.com
+            serviceName: gateway
+            servicePort: 8080
+            path: /
+```
+
+
+```sh
+$ helm upgrade openfaas \
+    --namespace openfaas \
+    ----reuse-values \
+    --values tls.yml \
+    openfaas/openfaas
+```
+
+## Create a certificate
+
+Finally, we can create the Certificate resource which triggers the actual creation of the certificate by `cert-manager`, edit the file below and replace the text `openfaas.mydomain.com` with your address for the API Gateway.
+
+```yaml
+# openfaas-crt.yaml
+apiVersion: certmanager.k8s.io/v1alpha1
+kind: Certificate
+metadata:
+  name: openfaas-crt
+spec:
+  secretName: openfaas-crt
+  dnsNames:
+    - openfaas.mydomain.com
+  acme:
+    config:
+      - http01:
+          ingressClass: nginx
+        domains:
+          - openfaas.mydomain.com
+  issuerRef:
+    name: letsencrypt-staging
+    kind: ClusterIssuer
+```
+
+```sh
+$ kubectl apply -f openfaas-crt.yaml
+```
+
+You can validate that certificate has been obtained successfully using
+
+```sh
+$ kubectl describe certificate openfaas-crt
+```
+
+If it was successful you can change to the production Let's Encrypt issuer by replacing `letsencrypt-staging` with `letsencrypt-prod` in the Certificate object.
```

Comment:
Perhaps tell them how to do this in a few bullets or which steps have to be re-run?